Chatify/chat/static/js/chat.js

```diff
@@ -1,80 +1,87 @@
+	var ws = new WebSocket('ws://127.0.0.1:8000/ws/chat/')
+
+	ws.onopen = function() {
+		}
+	ws.onmessage = function(e) {
+
+		let userDetail = JSON.parse(e.data)
+		if (userDetail.status == "offline") {
+			console.log($scope.chatData)
+			for (i = 0; i <= $scope.chatData.length; i++) {
+				console.log($scope.chatData[i].id)
+				if ($scope.chatData[i].id == userDetail.user_id) {
+					$scope.chatData.pop(userDetail.user_id)
+				}
+			}
+		}
+	}
+
+	function setUserStatus(status) {
+		ws.send(JSON.stringify({
+			'status': status
+		}))
+	}
+
+	ws.onclose = function(event) {
+	}
+
+
+
+	$scope.currentUser = undefined;
+	$scope.status = 'online';
+	$scope.msgText = {
+		text: ""
+	}
 
-    $scope.currentUser = undefined;
-    $scope.status = 'online';
-    $scope.msgText = {
-        text: ""
-    }
-
-    $scope.chatData = [{
-            id: 1,
-            name: "Karan",
-            profile: "http://127.0.0.1:8000/static/images/social_media.jpg",
-            messages: {
-                sender: [{
-                    user: "jaydip",
-                    profile: "http://127.0.0.1:8000/static/images/nature.jpeg",
-                    message: "hiiii karan this is jaydip [const data]",
-                    id: 2
-                }],
-                receiver: [{
-                    user: "karan",
-                    profile: "http://127.0.0.1:8000/static/images/social_media.jpg",
-                    message: "hiiii karan this is receiver karan",
-                    id: 1
-                }]
-            }
-        },
-        {
-            id: 2,
-            name: "jaydip",
-            profile: "http://127.0.0.1:8000/static/images/nature.jpeg",
-            messages: {
-                sender: [{
-                    user: "jaydip",
-                    profile: "http://127.0.0.1:8000/static/images/nature.jpeg",
-                    message: "hiiii karan this is jaydip",
-                    id: 2
-                }],
-                receiver: [{
-                    user: "karan",
-                    profile: "http://127.0.0.1:8000/static/images/social_media.jpg",
-                    message: "hiii karan this is receiver karan",
-                    id: 1
-                }]
-            }
-        }
-
-    ]
-
-    $scope.showChat = function(user) {
-        $scope.currentUser = user
-
-    };
-
-    $scope.sendChat = function(user) {
-        var message = $scope.msgText.text;
-        var currentUser = $scope.chatData.find(function(u) {
-            return u.id === user;
-        });
-        if (currentUser) {
-            currentUser.messages.sender.push({
-                user: currentUser.name,
-                profile: currentUser.profile,
-                message: message,
-            });
-        }
-    }
-    $scope.setStatus = function(status) {
-        $scope.status = status;
-    }
+	$scope.ajaxGet = function(url, callback = null) {
+		$http.get(url).then(function(response) {
+			if (callback) {
```

Comment:
multiple times added same comments don't use $http.get

Reply:
I saw your comment on Ajaxget, sir, but when I wrote this function, the get method wasn't functioning properly, which is why.

ajaxGet('GET',"/chat/api/userdata/", function(response) {
    $scope.chatData = response.UserData;
    console.log($scope.chatData)
    console.log("####")
})